ledger/chainsync.go

```diff
@@ -402,6 +403,16 @@ func (ls *LedgerState) processEpochRollover(
 	if err := ls.loadEpochs(txn); err != nil {
 		return fmt.Errorf("load epochs: %w", err)
 	}
+	// Update the scheduler interval based on the new epoch's slot length
+	if ls.Scheduler != nil {
+		slotLength := ls.currentEpoch.SlotLength
+		maxSafe := uint(math.MaxInt64 / time.Millisecond)
+		if slotLength > maxSafe {
+			return fmt.Errorf("SlotLength %d too large; overflows time.Duration", slotLength)
+		}
+		interval := time.Duration(int64(slotLength)) * time.Millisecond
```

Comment:
The slot length can be less than a second. In our `devnet` [configuration](https://github.com/blinklabs-io/docker-cardano-configs/blob/5ba968a40716046c189600b482e93582ab7cf798/config/devnet/shelley-genesis.json#L4), it's `0.1` (seconds).